### PR TITLE
Supporting multiple clusters for DAC validation

### PR DESCRIPTION
**Pull request by yahortsaryk**
## Description

This PR enables DAC validation for all DDC Clusters in the `Activated` status, which can be obtained only after passing the public referendum.

It also disables the strict requirement on the number of expected responses from DAC nodes and lets the OCW Validation cycles proceed if some Aggregators for the validating era are inaccessible due to networking issues or downtime.

Lastly, it fixes the problem for newly-joined validators and allows them to start validating process for eras newer than the last validated era for a cluster.

## Types of Changes
Please select the branch type you are merging and fill in the relevant template.

- [ ] Hotfix
- [ ] Release
- [x] Fix or Feature

### Types of Changes

- [x] Tech Debt (Code improvements)
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Dependency upgrade (A change in substrate or any 3rd party crate version)

### Migrations and Hooks

- [x] This change requires a runtime migration.
- [ ] Modifies `on_initialize`
- [ ] Modifies `on_finalize`

### Checklist for Fix or Feature

- [x] Change has been tested locally.
- [x] Change adds / updates tests if applicable.
- [ ] Changelog doc updated.
- [x] `spec_version` has been incremented.
- [ ] `network-relayer`'s [events](https://github.com/Cerebellum-Network/network-relayer/blob/dev-cere/shared/substrate/events.go) have been updated according to the blockchain events if applicable.
- [ ] All CI checks have been passed successfully

## Checklist for Hotfix

- [ ] Change has been deployed to Testnet.
- [ ] Change has been tested in Testnet.
- [ ] Changelog has been updated.
- [ ] Crate version has been updated.
- [ ] `spec_version` has been incremented.
- [ ] Transaction version has been updated if required.
- [ ] Pull Request to `dev` has been created.
- [ ] Pull Request to `staging` has been created.
- [ ] `network-relayer`'s [events](https://github.com/Cerebellum-Network/network-relayer/blob/dev-cere/shared/substrate/events.go) have been updated according to the blockchain events if applicable.
- [ ] All CI checks have been passed successfully

## Checklist for Release

- [ ] Change has been deployed to Devnet.
- [ ] Change has been tested in Devnet.
- [ ] Change has been deployed to Qanet.
- [ ] Change has been tested in Qanet.
- [ ] Change has been deployed to Testnet.
- [ ] Change has been tested in Testnet.
- [ ] Changelog has been updated.
- [ ] Crate version has been updated.
- [ ] Spec version has been updated.
- [ ] Transaction version has been updated if required.
- [ ] All CI checks have been passed successfully

### CODE REVIEW

File: pallets/ddc-verification/src/lib.rs [hunk 1]
```diff
@@ -3942,20 +4000,8 @@ pub mod pallet {
...
-		) -> DispatchResult {
-			ensure_root(origin)?;
-			ClusterToValidate::<T>::put(cluster_id);
-
-			Ok(())
-		}
-
 		// todo! Need to remove this
-		#[pallet::call_index(12)]
+		#[pallet::call_index(11)]
```
Comment (khssnv) [inline pallets/ddc-verification/src/lib.rs:4004]: Are you sure about the call index shift? Should we keep index `11` unused after the deletion?

File: pallets/ddc-verification/src/lib.rs [hunk 2]
```diff
@@ -2892,40 +2917,65 @@ pub mod pallet {
 		/// - `cluster_id`: cluster id of a cluster
 		/// - `dac_nodes`: List of DAC nodes
 		pub(crate) fn get_era_for_validation(
-			// todo! this needs to be rewriten - too complex and inefficient
 			cluster_id: &ClusterId,
 			dac_nodes: &[(NodePubKey, StorageNodeParams)],
 		) -> Result<Option<EraActivity>, OCWError> {
-			let current_validator_data = Self::fetch_current_validator()?;
+			let this_validator_data = Self::fetch_current_validator()?;
+			let this_validator = T::AccountId::decode(&mut &this_validator_data[..]).unwrap();
```
Comment (khssnv) [inline pallets/ddc-verification/src/lib.rs:2924]: Let's use `expect` instead of `unwrap` here and in the other updated places?

**Review by khssnv — APPROVED** [review]

File: pallets/ddc-verification/src/lib.rs (hunk 1 above)
Comment (yahortsaryk) [reply pallets/ddc-verification/src/lib.rs:4004]: As far as I know, any API change to extrinsics should be reflected by increasing `transaction_version` field in the chain spec. It should force UI clients to re-upload the chain metadata with updated indexes. I will try to test locally.

File: pallets/ddc-verification/src/lib.rs (hunk 2 above)
Comment (yahortsaryk) [reply pallets/ddc-verification/src/lib.rs]: Done

File: pallets/ddc-verification/src/lib.rs (hunk 1 above)
Comment (yahortsaryk) [reply pallets/ddc-verification/src/lib.rs:4006]: There is a good explanation in the comments to provided [answer](https://substrate.stackexchange.com/questions/8923/safe-way-to-remove-an-extrinsic).

**TLDR:** 
> There is nothing preventing you to reuse a certain extrinsic that was used in the past, but giving users some time to update on their side is a very healthy practice, I would say.

There are no users of the `ddc-verification` pallet extrinsics besides of Validators OCW. And we don't have any UI for that except the explorer that re-fetches the metadata once the `transaction_version` is updated. So it should be  safe
